### PR TITLE
Add color options to configuration

### PR DESCRIPTION
**Pull request by JordanRhode**

### CODE REVIEW

**Review by mdekrey — APPROVED** [review]
Looks great, thanks for providing this!

File: Server/BranchingStrategy/ColorConfiguration.cs [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+	private readonly ColorOptions colorOptions = colorOptions.Value;
+
 	public string DetermineColor(string branchName)
 	{
-		// TODO: move to somewhere with configuration
-		if (branchName.StartsWith("rc/")) return "rgb(111, 37, 111)";
-		if (branchName.StartsWith("line/") || branchName == "main" || branchName == "master") return "rgb(111, 206, 31)";
-		if (branchName.StartsWith("integ/") || branchName.StartsWith("integrate/") || branchName.StartsWith("integration/")) return "rgb(98, 98, 98)";
-		return "rgb(55, 127, 192)";
+		if (colorOptions.Branches.TryGetValue(branchName, out string? value))
```
Comment (chelfersps) [inline Server/BranchingStrategy/ColorConfiguration.cs:11]: Should we normalize the branch casing in either of the two usages here or are we certain they'll always match?

File: Server/BranchingStrategy/ColorConfiguration.cs [hunk 2]
```diff
@@ -1,13 +1,17 @@
-﻿namespace PrincipleStudios.ScaledGitApp.BranchingStrategy;
+﻿using Microsoft.Extensions.Options;
 
-public class ColorConfiguration : IColorConfiguration
+namespace PrincipleStudios.ScaledGitApp.BranchingStrategy;
+
+public class ColorConfiguration(IOptions<ColorOptions> colorOptions) : IColorConfiguration
```
Comment (chelfersps) [inline Server/BranchingStrategy/ColorConfiguration.cs:5]: Is a `[Theory]` unit test worth adding?

Comment (mdekrey) [reply Server/BranchingStrategy/ColorConfiguration.cs:5]: Yes, we should add tests - I should have caught this, too.

Comment (JordanRhode) [reply Server/BranchingStrategy/ColorConfiguration.cs:5]: I'll work on the tests for this after some meetings this AM, good call

File: Server/BranchingStrategy/ColorConfiguration.cs (hunk 1 above)
Comment (chelfersps) [reply Server/BranchingStrategy/ColorConfiguration.cs:11]: I should have phrased it, "are we certain we'll get the outcome we are looking for if we have a case mismatch", but I think you answered that

**Pull request merged**